src/hooks/sync-sites/sync-sites-context.tsx

```diff
@@ -99,6 +108,79 @@ export function SyncSitesProvider( { children }: { children: React.ReactNode } )
 	const { syncSites, isFetching, refetchSites } = useSyncSitesData();
 	useListenDeepLinkConnection( { refetchSites } );
 
+	const { client } = useAuth();
+	const { pushStatesProgressInfo } = useSyncStatesProgressInfo();
+	const hasInitialized = useRef( false );
+
+	// Initialize push states from in-progress server operations on mount
+	useEffect( () => {
+		// Only run once, when client and connectedSites are both available
+		if ( hasInitialized.current || ! client || connectedSites.length === 0 ) {
+			return;
+		}
+
+		hasInitialized.current = true;
+
+		const initializePushStates = async () => {
+			try {
+				// Get all local sites to map site IDs to SiteDetails
+				const allSites = await getIpcApi().getSiteDetails();
+
+				const restoredStates: PushStates = {};
+
+				// Query the server for each connected site to check for in-progress operations
+				for ( const connectedSite of connectedSites ) {
+					try {
+						// Find the local site
+						const selectedSite = allSites.find( ( site ) => site.id === connectedSite.localSiteId );
+
+						if ( ! selectedSite ) {
+							continue;
+						}
+
+						// Query the server for the current import status
+						const response = ( await client.req.get(
+							`/sites/${ connectedSite.id }/studio-app/sync/import`,
+							{
+								apiNamespace: 'wpcom/v2',
+							}
+						) ) as ImportResponse;
```

Comment:
```suggestion
						const response = await client.req.get< ImportResponse >(
							`/sites/${ connectedSite.id }/studio-app/sync/import`,
							{ apiNamespace: 'wpcom/v2' }
						);

```
The wpcom.js request functions support generics for the response body. This way, we avoid short-circuiting the type checker with the `as` operator.

Reply:
Thanks for the suggestion, much better now, added as part of 46bda48a